### PR TITLE
Styles: Add new CSS function mixins/presets for complex values

### PR DESCRIPTION
**Pull request by ItsJonQ**
This updates a handful fo CSS function mixins/presets to handful the (`string`) interpolation of complex CSS values, typically found in the `config.js` theme file.

The main function is `flow()`, which effectively combines `string` arguments together, and joined either with a space (` `) or a comma (`,`).

For a example, a `box-shadow` CSS value can be written with `flow`, like so:

```js
const boxShadow = flow(
	'0 1px',
	get('boxShadowSpreadValue'),
	'2px',
	get('boxShadowColor')
)
```

##### Combining groups

Groups (`Array<string>`) can be passed into `flow()`, which are combined and
comma separated. Useful for compounded CSS values (e.g. `box-shadow`).

```js
const boxShadow = flow([
		'0 1px',
		get('boxShadowSpreadValue'),
		'2px',
		get('boxShadowColor')
	], [
		'0 10px',
		get('boxShadowSpreadValue'),
		'20px',
		get('boxShadowColor')
	]
)
```

Addition functions include:

* `flow.calc()`

---

It's not necessary that all `box-shadow` or `calc()` or `rgb()` values are written using these functions. They are available as ways to make complex values easier to read and write.

---

### Types

I didn't add types for this PR. The reasons is because `create-styles` and `styles` are currently getting a types upgrade:
https://github.com/ItsJonQ/g2/pull/104

---

Resolves https://github.com/ItsJonQ/g2/issues/118

### CODE REVIEW

File: packages/styles/src/mixins/flow.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * 	], [
+ * 		'0 10px',
+ * 		get('boxShadowSpreadValue'),
+ * 		'20px',
+ * 		get('boxShadowColor')
+ * 	]
+ * )
+ * ```
+ *
+ * @param {...string|Array<string>} args CSS values to combine.
```
Comment (saramarcondes) [inline packages/styles/src/mixins/flow.js:38]:
```suggestion
 * @param {(string|string[])[]} args CSS values to combine.
```

File: packages/styles/src/mixins/flow.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * @returns {string} The combined CSS string value.
+ */
+export function flow(...args) {
+	const results = [];
```
Comment (saramarcondes) [inline packages/styles/src/mixins/flow.js:42]:
```suggestion
	/** @type {string[]} */
	const results = [];
```

File: packages/styles/src/presets/flow.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { flow as baseFlow } from '../mixins/flow';
+
```
Comment (saramarcondes) [inline packages/styles/src/presets/flow.js:2]:
```suggestion
/**
 * @type {{
	(...args: (string|string[])[]) => string;
	calc: (...args: (string|string[])[]) => string;
}}
 */
```

File: packages/components/src/__stories__/WIP/Presets.stories.js [hunk 4]
```diff
@@ -1,34 +1,31 @@
...
+				ui.padding(5),
+				ui.background.blue,
+				ui.scale(1),
+				ui.hover(ui.scale(1.5), ui.background.orange),
+				ui.animation.default,
+				{
+					boxShadow: ui.flow(
+						ui.calc(ui.get('controlHeight'), '+', ui.value.px(40)),
+						ui.get('gridBase'),
+						ui.calc('calc(1px + 20px)', '+', 'calc(1px + 20px)'),
```
Comment (saramarcondes) [inline packages/components/src/__stories__/WIP/Presets.stories.js:23]: Shouldn't this be `ui.flow.calc`? Where is `ui.calc` being set?

Comment (ItsJonQ) [reply packages/components/src/__stories__/WIP/Presets.stories.js:23]: Oh! Thank you 🙏 . My initial attempt at `flow` did indeed have `calc` at the main `ui` level. I changed it afterwards.

**Comment by ItsJonQ** [comment]
@saramarcondes Thank you for the types 😭 🙏

**Review by saramarcondes — APPROVED** [review]
LGTM!

**Pull request merged**